General/src/main/java/org/openimis/general/General.java

```diff
@@ -104,12 +104,14 @@ public String getFromRestApi(final String functionName) {
 		return content[0];
 	}
 
+	// ToDo: depricate this method if published to Google Play
 	public boolean isNewVersionAvailable(String Field,Context ctx, String PackageName){
 		String result = getFromRestApi("system/apkversion/" + Field);
 		if(result.contains(",")) {
 			result = result.replaceAll("(\\d+)\\,(\\d+)", "$1.$2");
 		}
-		return result == ""?false:Float.parseFloat(this.getVersion(ctx, PackageName).toString()) < Float.parseFloat(result);
+		// return result == ""?false:Float.parseFloat(this.getVersion(ctx, PackageName)) < Float.parseFloat(result);
+		return false;
```

Comment:
What is the goal of the function if it return false all the time ?

Reply:
I just modified this function because is verifying the version. As we have 3 number version X.Y.Z, it cannot be transformed directly in float (see commented line). In the near future we want to upload the mobile apps to Google Play, so the upgrade will be based on this. This function is to be removed in next releases.

Reply:
Ok but I don't like useless code, next time please comment all the code of the function if you return false anyway

another approach could be this (2nd answer https://stackoverflow.com/questions/198431/how-do-you-compare-two-version-strings-in-java)

import org.apache.maven.artifact.versioning.DefaultArtifactVersion;

DefaultArtifactVersion minVersion = new DefaultArtifactVersion("1.0.1");
DefaultArtifactVersion maxVersion = new DefaultArtifactVersion("1.10");

DefaultArtifactVersion version = new DefaultArtifactVersion("1.11");

if (version.compareTo(minVersion) < 0 || version.compareTo(maxVersion) > 0) {
    System.out.println("Sorry, your version is unsupported");
}